src/vendor/libbacktrace_2446c66076480ce07a6bd868badcbceb3eeecc2e_patch.diff

```diff
@@ -118,9 +115,120 @@ index bb481f3..2b056be 100644
 +			uintptr_t addr ATTRIBUTE_UNUSED,
 +			backtrace_syminfo_callback callback ATTRIBUTE_UNUSED,
 +			backtrace_error_callback error_callback, void *data);
-+
++extern int macho_add (struct backtrace_state *state, const char *filename, int descriptor,
++	   off_t offset, const unsigned char *match_uuid,
++	   uintptr_t base_address, int skip_symtab,
++	   backtrace_error_callback error_callback, void *data,
++	   fileline *fileline_fn, int *found_sym);
++extern void macho_syminfo (struct backtrace_state *state, uintptr_t addr,
++	       backtrace_syminfo_callback callback,
++	       backtrace_error_callback error_callback ATTRIBUTE_UNUSED,
++	       void *data);
++extern void macho_nosyms (struct backtrace_state *state ATTRIBUTE_UNUSED,
++	      uintptr_t addr ATTRIBUTE_UNUSED,
++	      backtrace_syminfo_callback callback ATTRIBUTE_UNUSED,
++	      backtrace_error_callback error_callback, void *data);
 +#ifdef __cplusplus
 +}
 +#endif
 +
  #endif
+diff --git a/macho.c b/macho.c
+index 66e101e..b6c19ba 100644
+--- a/macho.c
++++ b/macho.c
+@@ -92,6 +92,7 @@ struct macho_header_fat
+ 
+ #define MACH_O_MH_EXECUTE	0x02
+ #define MACH_O_MH_DYLIB		0x06
++#define MACH_O_MH_BUNDLE	0x08
+ #define MACH_O_MH_DSYM		0x0a
+ 
+ /* A component of a fat file.  A fat file starts with a
+@@ -278,6 +279,11 @@ struct macho_nlist_64
+ #define MACH_O_N_TYPE	0x0e	/* Mask for type bits */
+ #define MACH_O_N_STAB	0xe0	/* Stabs debugging symbol */
+ 
++#define MACH_O_N_FUN	0xe0	/* Stabs debugging symbol */
++#define MACH_O_GLOBAL   0x0f    /* global symbol */
++#define MACH_O_N_FNAME  0x26    /* static symbol: name,,n_sect,type,address */
++#define MACH_O_N_FUN    0x24    /* procedure: name,,n_sect,linenumber,address */
```

Comment:
Wait, this can't possibly work - you defined this macro on 152, and you're re-defining it again on 155, without an intervening `#undef`

Reply:
Yeah, the first macro should be removed